src/timesync/timesyncd-manager.c

```diff
@@ -1221,3 +1223,58 @@ static int manager_save_time_and_rearm(Manager *m, usec_t t) {
 
         return 0;
 }
+
+static const char* ntp_server_property_name[] = {
+        "SystemNTPServers",
+        "FallbackNTPServers",
+        "LinkNTPServers",
+        "RuntimeNTPServers",
+};
+
+static int on_deferred_ntp_server(sd_event_source *s, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        int r;
+        log_warning("doujb: here come to emit the ntp server signal function");
+        m->deferred_ntp_server_event_source = sd_event_source_disable_unref(m->deferred_ntp_server_event_source);
+        for(int type = SERVER_SYSTEM; type <= SERVER_RUNTIME; type++)
+        {
+                log_warning("doujb: here to emit the ntp server signal type = %d", type);
+                if (m->ntp_server_change_mask & (1U << type) ) {
+                        r = sd_bus_emit_properties_changed(
+                                m->bus,
+                                "/org/freedesktop/timesync1",
+                                "org.freedesktop.timesync1.Manager",
+                                ntp_server_property_name[type], NULL);
```

Comment:
please emit a single properties changed message for all server types that change at once. i.e. build up an strv, and then issue sd_bus_emit_properties_changed_strv